src/jit/compiler.cpp

```diff
@@ -11097,3 +11097,831 @@ bool Compiler::killGCRefs(GenTree* tree)
 
     return false;
 }
+
+#ifdef DEBUG
+//------------------------------------------------------------------------
+//                      VariableLiveRanges dumpers
+//------------------------------------------------------------------------
+
+// Dump "VariableLiveRange" when code has not been generated and we don't have so the assembly native offset
+// but at least "emitLocation"s and "siVarLoc"
+void dumpVariableLiveRange(const VariableLiveRange* varLiveRange, const CodeGenInterface* codeGen)
+{
+    // Check pointers parameter are not nullptr
+    noway_assert(codeGen != nullptr);
+    noway_assert(varLiveRange != nullptr);
+
+    codeGen->dumpSiVarLoc(&varLiveRange->m_VarHome);
+    printf(" [ ");
+    varLiveRange->m_StartEmitLocation.Print();
+    printf(", ");
+    if (varLiveRange->m_EndEmitLocation.Valid())
+    {
+        varLiveRange->m_EndEmitLocation.Print();
+    }
+    else
+    {
+        printf("NON_CLOSED_RANGE");
+    }
+    printf(" ]; ");
+}
+
+// Dump "VariableLiveRange" when code has been generated and we have the assembly native offset of each "emitLocation"
+void dumpVariableLiveRange(const VariableLiveRange* varLiveRange, emitter* _emitter, const CodeGenInterface* codeGen)
+{
+    // Check pointers parameter are not nullptr
+    noway_assert(varLiveRange != nullptr);
+    noway_assert(_emitter != nullptr);
+    noway_assert(codeGen != nullptr);
+
+    // "VariableLiveRanges" are created setting its location ("m_VarLocation") and the initial assembly offset
+    // ("m_StartEmitLocation")
+    codeGen->dumpSiVarLoc(&varLiveRange->m_VarHome);
+
+    // If this is an open "VariableLiveRange", "m_EndEmitLocation" is non-valid and print -1
+    UNATIVE_OFFSET endAssemblyOffset =
+        varLiveRange->m_EndEmitLocation.Valid() ? varLiveRange->m_EndEmitLocation.CodeOffset(_emitter) : -1;
+
+    printf(" [%X , %X )", varLiveRange->m_StartEmitLocation.CodeOffset(_emitter),
+           varLiveRange->m_EndEmitLocation.CodeOffset(_emitter));
+}
+
+//------------------------------------------------------------------------
+//                      LiveRangeDumper
```

Comment:
I have reservations about adding an additional, debug only, class just for 2 data members and one `if` statement (because that's pretty much the only interesting logic in this class).

Reply:
It is also taking responsibilities that VariableLiveDescriptor doesn't need to have, like knowing which of the VariableLiveRanges have been inserted during the last Basic Block.